stdlib/logging/config.pyi

```diff
@@ -15,9 +16,48 @@ if sys.version_info >= (3, 7):
 else:
     _Path = StrPath
 
+_TKey = TypeVar("_TKey")
+_TValue = TypeVar("_TValue")
+
 DEFAULT_LOGGING_CONFIG_PORT: int
 RESET_ERROR: int  # undocumented
 IDENTIFIER: Pattern[str]  # undocumented
+dictConfigClass = DictConfigurator
+
+class ConvertingMixin(Generic[_TKey, _TValue]):
+    def convert_with_key(self, key: _TKey, value: _TValue, replace: bool = ...) -> Dict[_TKey, _TValue]: ...
+    def convert(self, value: _TValue) -> Dict[_TKey, _TValue]: ...
+
+class ConvertingDict(Generic[_TKey, _TValue], ConvertingMixin[_TKey, _TValue]):
+    def __getitem__(self, key: _TKey) -> _TValue: ...
+    def get(self, key: _TKey, default: Optional[_TKey] = ...) -> Union[_TValue, _TKey]: ...
+    def pop(self, key: _TKey, default: Optional[_TKey] = ...) -> Union[_TValue, _TKey]: ...
+
+class ConvertingList(Generic[_TValue], ConvertingMixin[int, _TValue]):
+    @overload
+    def __getitem__(self, key: int) -> _TValue: ...
+    @overload
+    def __getitem__(self, key: slice) -> Sequence[_TValue]: ...
+    def pop(self, idx: int = ...) -> _TValue: ...
+
+class ConvertingTuple(Generic[_TValue], ConvertingMixin[Any, _TValue]):
+    @overload
+    def __getitem__(self, key: int) -> _TValue: ...
+    @overload
+    def __getitem__(self, key: slice) -> Tuple[_TValue]: ...
+
+class DictConfigurator(BaseConfigurator[_TKey, _TValue]):
+    def configure(self) -> None: ...
+    def configure_formatter(self, config: Dict[str, Dict[str, _TValue]]) -> Union[_TValue, Formatter]: ...
```

Comment:
I think this should just return `Formatter`. It can return other stuff if you put the right key/value in your dict, but the intended use case is probably a subclass of `Formatter`.